src/dmd/dsymbol.d

```diff
@@ -2146,3 +2135,31 @@ extern (C++) final class DsymbolTable : RootObject
         return cast(uint)dmd_aaLen(tab);
     }
 }
+
+
+/**********************************
+* Checks whether the current scope (or any of its parents) is deprecated.
+*
+* Params:
+*     sc = scope to check for
+*
+* Returns: `true` if any parent scope is deprecated, `false` otherwise`
+*/
+final bool isInDeprecatedScope(Scope* sc)
```

Comment:
In general I prefer not to split things out into functions unless they have more than one caller, or the caller is excessively large.  Also, in general, I think this refactoring should be in a separate PR as it is not essential to the bug-fix at hand, and should be argued on its own individual merits.

Reply:
Another problem with this method is it's modifying the public interface of the class, yet there is no public caller or potential caller.  If it is decided to keep this, it should probably be in its own private static function within the module.

Reply:
There are two callers: `checkComplexTransition` and `checkDeprecated` - it just felt wrong to duplicate code that has a certain functionality.

Reply:
Gah! Sorry.  Totally missed that.  You're right on here.

Reply:
I believe this also needs to be reflected in the C++ interface dsymbo.h